ethereumj-core/src/main/java/org/ethereum/net/rlpx/MessageCodesResolver.java

```diff
@@ -34,12 +36,21 @@ public void init(List<Capability> caps) {
         int offset = P2pMessageCodes.USER.asByte() + 1;
 
         for (Capability capability : caps) {
+
             if (capability.getName().equals(Capability.ETH)) {
                 setEthOffset(offset);
                 EthVersion v = fromCode(capability.getVersion());
                 offset += EthMessageCodes.values(v).length;
             }
 
+            if (capability.getName().equals(Capability.PAR)) {
+                // TODO: Why???
+                setParOffset(33);
```

Comment:
Message IDs are assigned according to the spec at https://github.com/ethereum/wiki/wiki/%C3%90%CE%9EVp2p-Wire-Protocol#payload-contents
33 comes from ETH/63 currently having 17 messages with IDs 0x10 - 0x20 and that `ETH` < `PAR` when compared in alphabetic order. Note that IDs should are assigned at session negotiating time for mutually supported subprotocols only.